### PR TITLE
fix scripts

### PR DESCRIPTION
**Pull request by kofi2718**
only linux

### CODE REVIEW

File: scripts/run.sh [hunk 1]
```diff
@@ -1,12 +1,19 @@
-#!/bin/sh
+#/bin/bash
```
Comment (D3SOX) [inline scripts/run.sh:1]: It should be `#!/usr/bin/bash` or better `#!/usr/bin/env bash`

**Comment by CanadaHonk** [comment]
Scripts have no problem, these are all environment specific changes.

**Comment by kofi2718** [comment]
@CanadaHonk That makes no sense. Why would anyone write them like that? Besides it make the script more universal for any linux environment than the original one
run.sh is written like it was meant to be in repo /
run.sh also executes scripts/injectPolyfills.sh, which makes no sense since injectPolyfills.sh is written in a way which implicates it should be in / just as run.sh should

Current run.sh file is environment specific, so it should be deleted or fixed to a more universal one.

**Comment by CanadaHonk** [comment]
They should be both be ran in the repo root, and work together fine. Changing sh -> bash is not needed and causes more potential issues. The changes also have mistakes like a missing ! in the shebang edited.

**Comment by CanadaHonk** [comment]
There are no npm dependencies in `src/package.json`, look at it yourself. Both the scripts are in the same directory, it does `./scripts/` as `.` is cwd, not where the script is. The scripts are fine and all the PR does is make the path configurable, which people can already do by modifying it. The scripts are for quality of life and are not required so they aren't a priority. The PR isn't mergeable and doesn't have any improvements so it won't be merged.

**Comment by kofi2718** [comment]
I see, you are right, however in this case the script should use $0 and **dirname** + /injectPolyfills.sh, so it always uses the right directory.

Also my app.asar path customization could **save time and** make it clear that they must supply the script with tier own environment specific path, as you said the scripts are for quality of life and i believe my scripts are a **quality of life enhancement** (excluding `./injectPolyfills.sh` which should be fixed)

`The scripts are for quality of life and are not required so they aren't a priority.` Does it mean they should never be updated under any circumstances? What kind of a point is this? I understand that they should be fix, but does it mean that **if i fix them they wont be merged** because of that?

Additionally the fact that one must modify the path in that script **isn't clear**, because the error message like this:
cp: cannot create directory ‘/etc/nonexistent/app.asar': No such file or directory
would be **hard to notice** since the script continues even if cp fails and the script would run `discord-canary "$@" # Run it`, which with original canary app.asar would **spam the console and make the error disappear**

**Comment by CanadaHonk** [comment]
It means I don't take it as a priority and won't merge PRs which aren't needed as they are already working as intended and only add on complexity and mistakes, they are mostly nitpicks anyways.

**Comment by kofi2718** [comment]
** add on complexity and mistakes** Yes. That's called improvement, things get more complicated(which in this case is like one if statement...) and handle more scenarios over time(get more universal and are easier to use).
besides if they are of such low priority: why do you care? Just don't write updates for those scripts and their complexity won't affect you, and if something stops working you can just revert the changes of that file

**PRs which aren't needed** ??? none of them are, they are meant to improve something

If you have some kind of unspoken needed/not-needed policy how about you write about it in the contributing guidelines?

**Comment by CanadaHonk** [comment]
This PR isn't an improvement, it breaks the shebang and (in my opinion) worsens the script by forcing you to specify a path every time instead of just being able to specify it once by editing the script itself, it is meant to be edited. The updates affect me as I'm the main person using the script.

**Comment by kofi2718** [comment]
Does it make it impossible to edit the script?
also you don't need to use my script, you can just use your old script. just don't `git add` the script once you modified it.
And yes, this script is broken, but i won't fix it if you wont approve it anyways. I was planning to, however apparently there is no point in doing so anyways

**Comment by CanadaHonk** [comment]
> Does it make it impossible to edit the script?

No, the point was it was meant to be edited, which you broke by forcing to use an argument.

You also didn't address how this will completely break the CI and existing people's workflows.

> also you don't need to use my script

Please realise this is my repo and changes internally will 99% effect me. You can still use your script you don't need to PR.

**Comment by kofi2718** [comment]
You seem to miss all of my good points and just see the bad in them. If you just add `|| exit 1` after cp i wont affect you and it will save time for enviroments with different paths

**Comment by CanadaHonk** [comment]
I understand some of your points but you aren't taking into account other people who are already using it like me, others, and the CI. I'm talking about the actual changes in the PR and not the points.

Locking this as it is unimportant and just becoming an unneeded argument.

**Pull request closed without merging**